dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/KernelFunctionHelpers/KernelFunctionHelpers.cs

```diff
@@ -0,0 +1,220 @@
+            {
+                throw new KernelException($"Parameter {param.Name} is required for function {functionMetadata.Name}.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// Processes the positional arguments passed to a Handlebars helper function.
+    /// </summary>
+    /// <param name="functionMetadata">KernelFunctionMetadata for the function being invoked.</param>
+    /// <param name="executionContext">Arguments maintained in the executing context.</param>
+    /// <param name="handlebarsArguments">Arguments passed to the Handlebars helper.</param>
+    /// <exception cref="KernelException">Thrown when a required parameter is missing.</exception>
+    private static void ProcessPositionalArguments(KernelFunctionMetadata functionMetadata, KernelArguments executionContext, Arguments handlebarsArguments)
+    {
+        var requiredParameters = functionMetadata.Parameters.Where(p => p.IsRequired).ToList();
+        if (requiredParameters.Count <= handlebarsArguments.Length && handlebarsArguments.Length <= functionMetadata.Parameters.Count)
+        {
+            var argIndex = 0;
+            foreach (var arg in handlebarsArguments)
+            {
+                var param = functionMetadata.Parameters[argIndex];
+                if (IsExpectedParameterType(param, arg))
+                {
+                    executionContext[param.Name] = handlebarsArguments[argIndex];
+                    argIndex++;
+                }
+                else
+                {
+                    throw new KernelException($"Invalid parameter type for function {functionMetadata.Name}. Parameter {param.Name} expects type {param.ParameterType ?? (object?)param.Schema} but received {handlebarsArguments[argIndex].GetType()}.");
+                }
+            }
+        }
+        else
+        {
+            throw new KernelException($"Invalid parameter count for function {functionMetadata.Name}. {handlebarsArguments.Length} were specified but {functionMetadata.Parameters.Count} are required.");
+        }
+    }
+
+    /// <summary>
+    /// Initializes the variables in the SK function context with the variables maintained by the Handlebars context.
+    /// </summary>
+    /// <param name="arguments">Arguments maintained in the executing context.</param>
+    private static KernelArguments GetKernelArguments(KernelArguments arguments)
+    {
+        var executionContext = new KernelArguments();
+        foreach (var v in arguments)
+        {
+            var value = v.Value ?? "";
+            var type = value.GetType();
+
+            var varString = !(type.IsPrimitive || type == typeof(string)) ? JsonSerializer.Serialize(value) : value.ToString();
+            executionContext.Add(v.Key, varString);
+        }
+
+        return executionContext;
+    }
+
+    /// <summary>
+    /// Invokes an SK function and returns a typed result, if specified.
+    /// </summary>
+    private static object? InvokeKernelFunctionAsync(
```

Comment:
Technically, it contains a synchronous wait when invoking the function since Handlebars doesn't support async operations

But I changed this back to `InvokeKernelFunction`